core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -194,12 +194,26 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
     endpoints.containsKey(name)
   }
 
+  def getNumOfThreads(conf: SparkConf): Int = {
+    val executorId = conf.get("spark.executor.id", "")
+    val isDriver = executorId == SparkContext.DRIVER_IDENTIFIER ||
+      executorId == SparkContext.LEGACY_DRIVER_IDENTIFIER
+    val side = if (isDriver) "driver" else "executor"
```

Comment:
This is also wrong. `RpcEnv` is used in other places - e.g. `Master` and `Worker` daemons.

Reply:
@vanzin , see my last comment for you. Besides, master and worker are relative more decoupled than driver and executor. We have alternatives to make them have different configurations.